### PR TITLE
Add OpenFaaS tutorial

### PR DESCRIPTION
**Pull request by puneetse**
This PR adds a tutorial for running OpenFaaS on Clear Linux

Note the GH preview file does not render all codeblocks properly. View the raw or build content locally.

### CODE REVIEW

**Comment by puneetse** [comment]
@qzheng527 can you review this PR? Feel free to comment here or over email.

File: source/tutorials/openfaas.rst [hunk 1]
```diff
@@ -0,0 +1,363 @@
+.. _openfaas:
+
+OpenFaaS\*
+##########
+
+This tutorial shows how to set up OpenFaaS running on top of a Kubernetes\*
+cluster on |CL|, obtain |CL| based OpenFaaS templates, and develop an example
```
Comment (qzheng527) [inline source/tutorials/openfaas.rst]: What is the link to CL? Or it is just a short name for Clear Linux?

File: source/tutorials/openfaas.rst [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+Prerequisites
+*************
+
+For simplicity, this tutorial assumes you have a Kubernetes single node
+cluster with only master node running |CL|.
+
+- For detailed instructions on how to install |CL|, see the :ref:`getting
+  started <get-started>` section.
+
+- For a detailed guide on how to set up Kubernetes, see the documentation on
```
Comment (qzheng527) [inline source/tutorials/openfaas.rst]: The document link seems got lost. I couldn't open it.

File: source/tutorials/openfaas.rst [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+      kubectl -n openfaas create secret generic basic-auth \
+      --from-literal=basic-auth-user=$FAAS_USER \
+      --from-literal=basic-auth-password="$FAAS_PASSWD"
+
+   .. code:: bash
+
+      kubectl apply -f faas-netes/yaml/
+
+   Wait for the OpenFaaS pods and services to get ready. This involves
+   downloading container images from the Internet and mayt takes some time
```
Comment (qzheng527) [inline source/tutorials/openfaas.rst]: It is "may" not "mayt", typo I guess

File: source/tutorials/openfaas.rst (hunk 1 above)
Comment (puneetse) [reply source/tutorials/openfaas.rst]: Yes, it's a replacement variable that gets substituted with the full branding name. You'll have to build the docs locally to see that.

File: source/tutorials/openfaas.rst (hunk 2 above)
Comment (puneetse) [reply source/tutorials/openfaas.rst]: This is rst syntax to refer to other local docs. It'll point to https://docs.01.org/clearlinux/latest/tutorials/kubernetes.html when it is built locally.

File: source/tutorials/openfaas.rst (hunk 3 above)
Comment (puneetse) [reply source/tutorials/openfaas.rst]: Thanks! I've fixed it.

File: source/tutorials/openfaas.rst [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+      model-downloader --name $MODEL_NAME -o $MODEL_DIR
+      model-converter --name $MODEL_NAME -d $MODEL_DIR -o $MODEL_DIR --mo $MO_PATH
+
+
+#. With the requirements added to the template. Write a python in the
+   :file:`handler.py` file. This function will parse the input picture URL,
+   find the model path, and call “classification_sample_async” to do image
+   classification. 
+
+   .. code-block:: python
```
Comment (mvincerx) [inline source/tutorials/openfaas.rst]: In running a test build of this PR, I see the python functions need proper indentation below each `def function():` . Will you please apply those changes? I assume the user wants to cut and paste your code-blocks.

File: source/tutorials/openfaas.rst (hunk 2 above)
Comment (mvincerx) [reply source/tutorials/openfaas.rst]: @puneetse. Is the dependency on the k8s tutorial, in its current state, a liability or concern?

Comment (puneetse) [reply source/tutorials/openfaas.rst]: @mvincerx I do not want to block this on the K8s improvements.

File: source/tutorials/openfaas.rst (hunk 4 above)
Comment (puneetse) [reply source/tutorials/openfaas.rst]: Thank you, the spacing did indeed get lost. I've updated it.

**Comment by puneetse** [comment]
I've pushed an update adding images. Waiting for a final +1 from @qzheng527 who is doing another validation pass. After that, this can be merged from my perspective.

**Comment by mvincerx** [comment]
@puneetse. Thanks for the update. I'm watching this PR closely. 😉

**Comment by qzheng527** [comment]
LGTM

**Comment by mvincerx** [comment]
LGTM. Thanks, all.

**Pull request merged**